clang_tidy/clang_tidy.bzl

```
@@ -56,9 +56,15 @@ def _run_tidy(ctx, exe, flags, compilation_context, infile, discriminator):
 
 def _rule_sources(ctx):
     srcs = []
+    print("input source: ", ctx.rule.attr.srcs)
+    file_extentions = ["c", "cc", "cpp", "cxx", "c++", "C", "h", "hh", "hpp", "hxx", "inc", "inl", "H"]
     if hasattr(ctx.rule.attr, "srcs"):
         for src in ctx.rule.attr.srcs:
-            srcs += [src for src in src.files.to_list() if src.is_source]
+            for path in src.files.to_list():
+                for file_extension in file_extentions:
+                    if file_extension == path.extension:
+                        print("path:", path)
```

嵌套了三层for循环+两层if，可以适当注意优化代码结构：
```suggestion
            for path in src.files.to_list():
                    if path.extension in file_extentions:
                            # do something
```